SpringMVCAnnotationOnlineStore/src/main/java/org/o7planning/springmvconlinestore/controller/MainController.java

```diff
@@ -158,19 +168,56 @@ public String processNewUser(HttpServletRequest request, //
     
     @RequestMapping({ "/buyProduct" })
     public String listProductHandler(HttpServletRequest request, Model model, //
-            @RequestParam(value = "code", defaultValue = "") int code) {
- 
+            @RequestParam(value = "code", defaultValue = "") int code, @ModelAttribute("productForm") //
+    		ProductInfo prodInfo, BindingResult result) {
+    		
+    	
         Product product = null;
         if (code > 0) {
             product = productDAO.findProduct(code);
         }
         if (product != null) {
- 
+        	Double totalPrice = 0.00;
+			
             // Cart info stored in Session.
             CartInfo cartInfo = Utils.getCartInSession(request);
  
             ProductInfo productInfo = new ProductInfo(product);
- 
+            
+            
+           //Update user selections
+            productInfo.setSize(request.getParameter("size"));
+            
+            if(request.getParameter("engraving") != null)
```

Comment:
@KendraMBach should we replace `null` with `0` and add a constraint to this table? `Null` works fine for class, but probably isn't good enough for the real world.

Reply:
We can chat about this today. But the ProductInfo model and Product entity aren’t the same. For the controller null is necessary for instantiation of certain ProductInfo variables. A specific value aside from null would cause more difficulties in the long run with checks.